docs/Shell-Completion.md

```diff
@@ -23,7 +23,8 @@ if type brew &>/dev/null; then
 fi
 ```
 
-Should you later install the `bash-completion` formula, this will automatically use its initialization script to read the completions files.
+Should you later install the `bash-completion@2` formula, this will automatically use its initialization script to read the completions files.
+When installed, the `bash-completion@2` formula also runs `${HOMEBREW_PREFIX}/etc/profile.d/bash_completion.sh` and all files in the `bash_completion.d` directory (If the Caveats section is followed after installation, of course). Therefore, it is recommended to comment out the block of code given above after installing `bash-completion@2` because it becomes redundant.
```

Comment:
I think it's better to recommend ignoring the caveat if the code above is used. The code above does the same as the caveat but also works if the formula isn't installed.

Reply:
That makes sense. But don't you think we should let the formula have control? For example, what if `bash-completion@2` decided to change which script it runs or adds more files to source in the Caveats code?

Reply:
They haven't for about 20 years, but if they do someone will have to remember to update this.

Reply:
I hope the new commit fixed this problem!